scripting/get5/stats.sp

```diff
@@ -517,8 +517,84 @@ static int GetClutchingClient(int csTeam) {
 public void DumpToFile() {
   char path[PLATFORM_MAX_PATH + 1];
   if (FormatCvarString(g_StatsPathFormatCvar, path, sizeof(path))) {
-    g_StatsKv.ExportToFile(path);
+    DumpToFilePath(path);
+  }
+}
+
+public bool DumpToFilePath(const char[] path) {
+  int path_length = strlen(path);
+  // Check if path ends in ".json"
+  if (path_length >= 5) {
+    char ext[5];
+    ext[0] = path[path_length-5];
+    ext[1] = path[path_length-4];
+    ext[2] = path[path_length-3];
+    ext[3] = path[path_length-2];
+    ext[4] = path[path_length-1];
+    if (StrEqual(".json", ext, false)) {
+      // Dump in json format
+      return DumpToJSONFile(path);
+    }
+  }
+
+  // Does not end with json, default to VKV
+  return g_StatsKv.ExportToFile(path);
+}
+
+public bool DumpToJSONFile(const char[] path) {
+  g_StatsKv.Rewind();
+  g_StatsKv.GotoFirstSubKey(false);
+  JSON_Object stats = EncodeKeyValue(g_StatsKv);
+  g_StatsKv.Rewind();
+
+  File stats_file = OpenFile(path, "w");
+  if (stats_file == null) {
+    LogError("Failed to open stats file");
+    return false;
   }
+
+  char jsonBuffer[8192]; // 8 KiB
```

Comment:
I'm not sure how large the json data is in practice TBH, I'm wondering if 8KiB is enough; do you have a sense of the size for a normal 10-person 3-map match?

Reply:
This was a guess based on intuition. I tried 32 KiB but that crashed with a out of heap or stack space error (I don't recall which)